src/instructlab/eval/unitxt.py

```diff
@@ -0,0 +1,127 @@
+"""
+Unitxt - Flexible, Shareable and Reusable Data Preparation and Evaluation for Generative AI
+https://github.com/IBM/unitxt
+https://arxiv.org/abs/2401.14019
+"""
+
+# Standard
+from uuid import uuid4
+import os
+import shutil
+
+# Third Party
+from lm_eval.tasks.unitxt import task
+import yaml
+
+# First Party
+from instructlab.eval.mmlu import MMLUBranchEvaluator
+
+# Local
+from .logger_config import setup_logger
+
+logger = setup_logger(__name__)
+
+TEMP_DIR_PREFIX = "unitxt_temp"
+
+
+class UnitxtEvaluator(MMLUBranchEvaluator):
+    """
+    An evaluator class, running Unitxt evaluation
+
+    Attributes:
+        model_path      absolute path to or name of a huggingface model
+        unitxt_recipe   unitxt recipe (see unitxt.ai for more information)
+                        A Recipe holds a complete specification of a unitxt pipeline
+                        Example: card=cards.wnli,template=templates.classification.multi_class.relation.default,max_train_instances=5,loader_limit=20,num_demos=3,demos_pool_size=10
+
+    """
+
+    name = "unitxt"
+
+    def __init__(
+        self,
+        model_path,
+        unitxt_recipe: str,
+    ):
+        unitxt_task = self.assign_task_name()
+        tasks_dir = self.assign_tasks_dir(unitxt_task)
+        super().__init__(
+            model_path=model_path, tasks_dir=tasks_dir, tasks=[unitxt_task], few_shots=0
+        )
+        self.unitxt_recipe = unitxt_recipe
+
+    def assign_tasks_dir(self, task_name):
+        return f"{TEMP_DIR_PREFIX}_{task_name}"
+
+    def assign_task_name(self):
+        return str(uuid4())
+
+    def prepare_unitxt_files(self) -> None:
+        taskname = self.tasks[0]
+        yaml_file = os.path.join(str(self.tasks_dir), f"{taskname}.yaml")
+        create_unitxt_pointer(self.tasks_dir)
+        create_unitxt_yaml(
+            yaml_file=yaml_file, unitxt_recipe=self.unitxt_recipe, task_name=taskname
+        )
+
+    def remove_unitxt_files(self):
+        if self.tasks_dir.startswith(
+            TEMP_DIR_PREFIX
+        ):  # to avoid unintended deletion if this class is inherited
+            shutil.rmtree(self.tasks_dir)
+        else:
+            logger.warning(
+                "unitxt tasks dir did not start with '%s' and therefor was not deleted",
+                TEMP_DIR_PREFIX,
+            )
+
+    def run(self, server_url: str | None = None) -> tuple:
+        """
+        Runs evaluation
+
+        Returns:
+            overall_scores       Average scores for the task group
```

Comment:
Misaligned comment

Reply:
The descriptions weren't aligned top to bottom.  Looks like it's fixed now, thanks!